R/bm-remote-dataset.R

```diff
@@ -0,0 +1,32 @@
+#' Remote (S3) dataset reading
+#'
+#' @export
+remote_dataset <- Benchmark("remote_dataset",
+  setup = function(source = c("taxi_file_list_parquet", "taxi_file_list_feather")) {
+    library("dplyr")
+    dataset <- ensure_dataset(source, download = FALSE)
+    result_dim <- get_dataset_attr(source, "dim")
+
+    BenchEnvironment(
+      dataset = dataset,
+      expected_dim = result_dim
+    )
+  },
+  before_each = {
+    options("arrow.use_async" = TRUE)
```

Comment:
This is something that we could make a parameter by adding it as an argument to the `setup` function above and passing the value through if it would be interesting to test these with async on and with it off.

Reply:
I want to leave it out so it is not a parameter.  It will disappear someday.  I'd like this all to (eventually) be something the user is not aware of so the story is simply "reading datasets got faster" and not "we introduced this new concept you need to be aware of".